pkg/kubelet/server/server.go

```diff
@@ -280,14 +287,34 @@ func (s *Server) InstallAuthFilter() {
 	})
 }
 
+// addMetricsBucketMatcher adds a regexp matcher and the relevant bucket to use when
+// it matches. Please be aware this is not thread safe and should not be used dynamically
+func (s *Server) addMetricsBucketMatcher(matcher *regexp.Regexp, bucket string) {
+	s.bucketMatchers = append(s.bucketMatchers, metricsBucket{matcher, bucket})
+}
+
+// getMetricBucket find the appropriate metrics reporting bucket for the given path
+func (s *Server) getMetricBucket(path string) string {
+	for _, metricsBucket := range s.bucketMatchers {
+		matcher, bucket := metricsBucket.matcher, metricsBucket.name
+		if matcher.MatchString(path) {
+			return bucket
+		}
+	}
+	return "Invalid path"
+}
+
 // InstallDefaultHandlers registers the default set of supported HTTP request
 // patterns with the restful Container.
 func (s *Server) InstallDefaultHandlers(enableCAdvisorJSONEndpoints bool) {
+	s.addMetricsBucketMatcher(regexp.MustCompile(`^healthz(/.*)?$`), "healthz")
```

Comment:
I think we'd be interested in path metrics even if the handler isn't installed. Given that, it may be easier to add all the bucket matchers in one place.

Reply:
I think its fine to add always desired buckets at the top. However I also want an indicator to someone adding a new handler (and is copying existing code), that they need to add a metric bucket.